### DISCUSSION TITLE
How can  I use rnnoise in my project ?

### DISCUSSION THREAD

**Discussion by yolin329** [wegylexy/rnnoise_wasm#11] [Q&A] [answered]
As the title, how can I use rnnoise in my project? Should I just put 3 files in folder "dist" in my project? But the console shows that RNNoiseNode is not found. I need some help. Thank you very much.

**Comment by wegylexy** [comment]
Please check out the demo folder for sample usage.

**Comment by yolin329** [comment]
Very thanks for your reply. But I didn't find any import / require script in demo code, and I put the files in "dist" in the same folder that I want to use the microphone, but the error shows that RNNoiseNode is not found.  Is that because I put the scripts in wrong place? Thank you.

![image](https://user-images.githubusercontent.com/52409634/142967104-ff646b10-5168-4106-88d6-9d99e731d301.png)

**Comment by wegylexy** [comment, marked as answer]
It is not a module. Just include it the traditional way before the first usage.
https://github.com/wegylexy/rnnoise_wasm/blob/776ca4945d4c0f51212c56d0652c9c3195a48661/demo/index.html#L37

**Comment by yolin329** [comment]
Yes, I include it like below in the script that deal with microphone: 
`require('./rnnoise-runtime.js')` 

But when I want to RNNoiseNode, the error show.
So now, I am stuck.

But anyway I'll keep trying, thank you very much.

**Comment by wegylexy** [comment]
No, it's `<script src="../dist/rnnoise-runtime.js"></script>` in HTML, not `import`/`require`.

Am I not clear enough? It is NOT a module.

**Reply by husainnazer** [reply]
Is there a way to import this as a module?

I'm using react and I have the same problem.

**Reply by wegylexy** [reply]
RNNoiseNode must be loaded in the global namespace, and then `react` can use it just like other built-in types.
If you use TypeScript, declare it.
What's your problem?

**Reply by husainnazer** [reply]
I added `<script src="../dist/rnnoise-runtime.js"></script>` inside index.html and this showed up in the console.

![Screenshot 2022-07-15 104606](https://user-images.githubusercontent.com/63315100/179155620-e499f98b-9512-435d-9526-36a3d86a8a38.png)

**Reply by wegylexy** [reply]
The file starts with `'use strict'`, not `<`, so your server must be serving something else.

**Reply by husainnazer** [reply]
My bad, I put the /dist folder inside /src but index.html is inside the /public folder which can't access the /src folder. 
So I placed the /dist folder inside /public folder and now it's working.

Thanks a lot.
